packages/chat/components/chat/src/chat.ts

```diff
@@ -0,0 +1,389 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { LitElement } from 'lit';
+import { property, state } from 'lit/decorators.js';
+import { LlamaPluginAPI } from '../../../services/LlamaPlugin/index.js';
+// @ts-ignore
+import styles from './chat.scss?inline';
+
+//const { stablePrefix: c4aiPrefix } = settings;
+
+/**
+ * Input component using search typeahead api
+ */
+export default class C4AIChat extends LitElement {
+  static styles = styles;
+
+  /**
+   * Core array of message objects to display and interact with, invidual messages are currently described as:
+   * {"message":"displayed text", "origin":"can be bot or user, depending on user-defined role names"}
+   */
+  private _messages: any[] = [];
+
+  /**
+   * string variable edited by textInput, auto-updates at every keystroke and is sent to the api url on 'enter' or 'send' button click
+   */
+  @state()
+  _messageText = '';
+
+  /**
+   * server side integer denoting the number of messages sent in total
+   */
+  private _eventNumber = 0;
+
+  /**
+   * session ID to ensure no two conversations collide within the server
+   */
+  private _session = '' + Math.random();
+
+  /**
+   * boolean denoting when an api query has begun and returned to 'false' when it is received or an error occured, used to display an empty loading message
+   */
+  @state()
+  _queryInProgress = false;
+
+  /**
+   * current string returned by the input dom object
+   **/
+  private _inputText = '';
+
+  /**
+   * string url denoting where the message query will be sent, either BAM or watsonx.ai or any other service
+   */
+  @property({ type: String, attribute: 'api-url' })
+  apiURL;
+
+  /**
+   * string denoting which model to use, only 'llama-2' is available currently
+   */
+  @property({ type: String, attribute: 'feedback-url' })
+  feedbackURL;
+
+  /**
+   * string denoting which model to use, only 'llama-2' is available currently
+   */
+  @property({ type: String, attribute: 'model' })
+  model;
+
+  /**
+   * float varying from 0.0 to 1.0, denotes how 'creative' the model's response will be. 0.0 (default) is the most safe and predictable while 1.0 is hightly creative but unpredictable (not advised for operations returning code or JSON objects)
+   */
+  @property({ type: Number, attribute: 'temperature' })
+  temperature;
+
+  /**
+   * string denoting the user name, used for internal logic in the server to differentiate bot responses and user reseponses. default: 'user' but should be the user's real name based on IBM ID or any other data available
+   */
+  @property({ type: String, attribute: 'username' })
+  username;
+
+  /**
+   * string denoting the bot name, default: 'bot' but can be changed to 'Watson' or 'client assistant' or any other name
+   */
+  @property({ type: String, attribute: 'agentname' })
```

Comment:
```suggestion
  @property({ type: String, attribute: 'agent-name' })
```

usually we seperate two words by a dash in properties. Anywhere `agentname` is used should be updated to `agent-name`